src/object.rs

```diff
@@ -1436,7 +1436,7 @@ impl<T: ObjectType> ObjectExt for T {
 
     fn downgrade(&self) -> WeakRef<T> {
         unsafe {
-            let w = WeakRef(Box::new(mem::uninitialized()), PhantomData);
+            let w = WeakRef(Box::new(mem::MaybeUninit::uninit().assume_init()), PhantomData);
```

Comment:
This is wrong. `assume_init()` can only be called after the `g_weak_ref_init()` really... but `assume_init()` involves moving data around and that would break the weakref.

I think this can simply be `mem::zeroed()`